cmd/kpod/stats.go

```diff
@@ -120,126 +110,117 @@ func getStats(server *libkpod.ContainerServer, args []string, all bool, statsCha
 		step = 0
 	}
 	for i := 0; i < times; i += step {
-		reportStats := []*libkpod.ContainerStats{}
+		reportStats := []*libpod.ContainerStats{}
 		for _, ctr := range ctrs {
 			id := ctr.ID()
 			if _, ok := containerStats[ctr.ID()]; !ok {
-				initialStats, err := server.GetContainerStats(ctr, &libkpod.ContainerStats{})
+				initialStats, err := ctr.GetContainerStats(&libpod.ContainerStats{})
 				if err != nil {
 					return err
 				}
 				containerStats[id] = initialStats
 			}
-			stats, err := server.GetContainerStats(ctr, containerStats[id])
+			stats, err := ctr.GetContainerStats(containerStats[id])
 			if err != nil {
 				return err
 			}
 			// replace the previous measurement with the current one
 			containerStats[id] = stats
 			reportStats = append(reportStats, stats)
 		}
-		statsChan <- reportStats
-
-		err := server.Update()
+		ctrs, err = containerFunc()
 		if err != nil {
 			return err
 		}
-		ctrs, err = server.ListContainers(isRunning, ctrInList(args))
-		if err != nil {
-			return err
+		if strings.ToUpper(format) != "JSON" && !c.Bool("no-reset") {
+			tm.Clear()
+			tm.MoveCursor(1, 1)
+			tm.Flush()
 		}
+		outputStats(reportStats, format)
+		time.Sleep(time.Second)
 	}
 	return nil
 }
 
-func outputStats(stats []*libkpod.ContainerStats, format string, json bool) error {
-	if format == "" {
-		outputStatsHeader()
+func outputStats(stats []*libpod.ContainerStats, format string) error {
+	var out formats.Writer
+	var outputStats []statsOutputParams
+	for _, s := range stats {
+		outputStats = append(outputStats, getStatsOutputParams(s))
 	}
-	if json {
-		return outputStatsAsJSON(stats)
+	if len(outputStats) == 0 {
+		return nil
 	}
-	var err error
-	for _, s := range stats {
-		if format == "" {
-			outputStatsUsingFormatString(s)
-		} else {
-			params := getStatsOutputParams(s)
-			err2 := outputStatsUsingTemplate(format, params)
-			if err2 != nil {
-				err = errors.Wrapf(err, err2.Error())
-			}
-		}
+	if strings.ToUpper(format) == "JSON" {
```

Comment:
Same thing here, can just compare to `JSONString`

Reply:
fixed